### PR TITLE
librats: add tdx event-log function and sample code for TDX and Occlum

### PR DESCRIPTION
**Pull request by zhiwei-intel-h**
Signed-off-by: Zhiwei Han <zhiwei1.han@intel.com>

### CODE REVIEW

File: attesters/tdx-ecdsa/collect_evidence.c [hunk 1]
```diff
@@ -35,6 +39,71 @@ static int tdx_get_report(const tdx_report_data_t *report_data, tdx_report_t *td
...
+	        /* TDEL is optional */
+	        return RATS_ATTESTER_ERR_NONE;
+	}
+
+	unsigned char tdel_info[TDEL_INFO_SZ];
+	int tdel_info_sz = read(fd, tdel_info, sizeof(tdel_info));
+	if (tdel_info_sz != sizeof(tdel_info)) {
+	        close(fd);
+	        RATS_INFO("failed to read TDEL info\n");
+	        return -RATS_ATTESTER_ERR_INVALID;
```
Comment (zeuson0) [inline attesters/tdx-ecdsa/collect_evidence.c:60]: `-RATS_ATTESTER_ERR_INVALID` -> `RATS_ATTESTER_ERR_INVALID`

File: attesters/tdx-ecdsa/collect_evidence.c [hunk 2]
```diff
@@ -35,6 +39,71 @@ static int tdx_get_report(const tdx_report_data_t *report_data, tdx_report_t *td
...
+	}
+
+
+	unsigned char tdel_data[TDEL_DATA_SZ];
+	int tdel_data_sz = read(fd, tdel_data, sizeof(tdel_data));
+	if (tdel_data_sz <= 0) {
+	        close(fd);
+	        RATS_INFO("failed to read TDEL data\n");
+	        return -RATS_ATTESTER_ERR_INVALID;
```
Comment (zeuson0) [inline attesters/tdx-ecdsa/collect_evidence.c:91]: `-RATS_ATTESTER_ERR_INVALID` -> `RATS_ATTESTER_ERR_INVALID`

File: attesters/tdx-ecdsa/collect_evidence.c [hunk 3]
```diff
@@ -96,10 +165,21 @@ rats_attester_err_t tdx_ecdsa_collect_evidence(rats_attester_ctx_t *ctx,
 
 	RATS_DEBUG("Succeed to generate the quote!\n");
 
+    int tdel_info_len = 0;
+    if (tdx_get_tdel_info(ctx, evidence, &tdel_info_len) != RATS_ATTESTER_ERR_NONE)
+            return -RATS_ATTESTER_ERR_INVALID;
```
Comment (zeuson0) [inline attesters/tdx-ecdsa/collect_evidence.c:170]: ditto

File: attesters/tdx-ecdsa/collect_evidence.c [hunk 4]
```diff
@@ -96,10 +165,21 @@ rats_attester_err_t tdx_ecdsa_collect_evidence(rats_attester_ctx_t *ctx,
...
+
+    /* TDEL information is optional */
+    int tdel_data_len = 0;
+    if (tdel_info_len && tdx_get_tdel_data(ctx, evidence, &tdel_data_len) != RATS_ATTESTER_ERR_NONE)
+            return -RATS_ATTESTER_ERR_INVALID;
```
Comment (zeuson0) [inline attesters/tdx-ecdsa/collect_evidence.c:175]: ditto

**Comment by haosanzi** [comment]
replaced by #79

**Pull request closed without merging**